producer.go

```diff
@@ -535,16 +535,25 @@ func (p *producer) dispatchWork(workCtx context.Context, count int, fetchResultC
 func (p *producer) heartbeatLogLoop(ctx context.Context) {
 	ticker := time.NewTicker(5 * time.Second)
 	defer ticker.Stop()
+	type state struct {
+		jobsRan    uint64
+		jobsActive int
+	}
+	prev := state{}
```

Comment:
And similarly here too. Maybe:

* `cur` -> `curStats`
* `prev` -> `prevStats`